lib/pageUtils.js

```diff
@@ -0,0 +1,24 @@
+export const scrollTo = (element, to, duration) => {
+  if (duration <= 0) return
+  const difference = to - element.scrollTop;
+  const perTick = difference / duration * 16
+
+  window.requestAnimationFrame(() => {
+    element.scrollTop = element.scrollTop + perTick
+    if (element.scrollTop === to) return
+    scrollTo(element, to, duration - 16)
+  })
+}
+
+export const urlQuery = ((a) => {
+  if (a == '') return {}
+  let b = {}
+  for (let i = 0; i < a.length; ++i) {
+    let p=a[i].split('=', 2)
+    if (p.length == 1)
+      b[p[0]] = ''
+    else
+      b[p[0]] = decodeURIComponent(p[1].replace(/\+/g, ' '))
+  }
+  return b
```

Comment:
You're not iterating over the length of a string but the items in an array with `window.location.search.substr(1).split('&')`.

```javascript
const padder = (p) => p.length < 2 ? [p[0], ''] : p
const reducer = (prev, curr, idx) => {
  const p = padder(curr.split("=", 2))
  prev[p[0]] = decodeURIComponent(p[1].replace(/\+/g, ' '))
  return prev
}
const query = (q) => q === '' ? {} : q.reduce(reducer, {})
```

No doubt you could write a more succinct reducer